app/services/facet.service.js

```diff
@@ -2,7 +2,24 @@ import angular from 'angular';
 
 angular.module('facetService', []).
 
+// Provides the ability to filter sets of data via various criteria.
+// The Facet service operates on lists of objects, and filters them
+// according to criteria which evaluates specific keys in those
+// objects.
+// Facets provide several different ways of matching:
+// * string   - if the test value is a string, then the facet only matches if
+//              the object has a value at that key which is an exact match.
+// * regexp   - if the test value is a regexp, then the facet matches if
+//              the object has a value at that key which passes value.match(regexp).
+// * function - if the test value is a function, then the facet matches if
+//              the function passes true when passed the value at that key from the object.
 factory('Facet', function() {
+  // Adds a new facet.
+  // * name - the key this function should match.
+  // * value - the value to match against the object's value; see above.
+  // * data
```

Comment:
I had a hard time remembering this. :sweat_smile: I'll look it up and document it.